src/components/CodeSnippet/CodeSnippet.stories.mdx

```diff
@@ -0,0 +1,144 @@
+  <Story name="Title">
+    <CodeSnippet blocks={[ { title: "Output", code: "Mode: all Settings: maas_url=http://192.168.122.1:5240/MAAS"} ]} />
+  </Story>
+</Canvas>
+
+### Multiple blocks
+
+<Canvas>
+  <Story name="Multiple blocks">
+    <CodeSnippet blocks={[
+      { title: "JavaScript", code: "console.log('Vanilla');"},
+      { title: "Output", code: "Vanilla" },
+    ]} />
+  </Story>
+</Canvas>
+
+
+### Icon / Prompt
+
+<Canvas>
+  <Story name="Icon">
+    <CodeSnippet blocks={[
+       { title: "Install on Linux", icon: "linuxPrompt", code: "snap install toto"},
+       { title: "Install on Windows", icon: "windowsPrompt", code: "snap install toto"},
+       { title: "Get from the Store", icon: "url", code: "http://snapcraft.io/toto"} ,
+     ]}/>
+  </Story>
+</Canvas>
+
+### Numbered
+
+<Canvas>
+  <Story name="Numbered">
+    <CodeSnippet blocks={[
+       { numbered: true,
+         code: `#!/bin/bash
+set -eu . $CONJURE_UP_SPELLSDIR/sdk/common.sh
+if [[ "$JUJU_PROVIDERTYPE" == "lxd" ]]; then
+  debug "Running pre-deploy for $CONJURE_UP_SPELL"
+  sed "s/##MODEL##/$JUJU_MODEL/" $(scriptPath)/lxd-profile.yaml | lxc profile edit "juju-$JUJU_MODEL" || exposeResult "Failed to set profile" $? "false"
+fi
+exposeResult "Successful pre-deploy." 0 "true"`
+       },
+     ]}/>
+  </Story>
+</Canvas>
+
+### Wrapped
+
+<Canvas>
+  <Story name="Wrapped">
+    <CodeSnippet blocks={[
+       { numbered: true,
+         isWrapped: true,
+         code: `#!/bin/bash
+set -eu . $CONJURE_UP_SPELLSDIR/sdk/common.sh
+if [[ "$JUJU_PROVIDERTYPE" == "lxd" ]]; then
+  debug "Running pre-deploy for $CONJURE_UP_SPELL"
+  sed "s/##MODEL##/$JUJU_MODEL/" $(scriptPath)/lxd-profile.yaml | lxc profile edit "juju-$JUJU_MODEL" || exposeResult "Failed to set profile" $? "false"
+fi
+exposeResult "Successful pre-deploy." 0 "true"`
+       },
+     ]}/>
+  </Story>
+</Canvas>
+
+### Dropdowns
+
+Dropdown select menus can be added to the headers of code blocks to allow users to choose one of the options. The dropdowns options are passed via `dropdowns` property in the block options object.
+
+Dropdown configuration object is structured as follows:
+<ArgsTable of={CodeSnippetDropdown} />
+
+With the `options` being an array of option properties compatible with the `Select` options (`[{ label: string, value: string | number }]`).
+
+<Canvas>
+  <Story name="Dropdowns">
+  {() => {
+    const [lang, setLang] = useState("html");
+    const code = {
+      js: `console.log("Example 1");`,
+      css: `.p-heading--two { color: red; }`,
+      html: `<h1 class="p-heading--two">How to use code snippets</h1>`,
+    };
+    return <CodeSnippet blocks={[
+       {
+         code: code[lang],
+         dropdowns: [
+           { options: [
+              { value: "js", label: "JS" },
+              { value: "css", label: "CSS" },
+              { value: "html", label: "HTML", selected: true },
+            ],
+            onChange: (event) => { setLang(event.target.value) }
```

Comment:
I feel like this switching should be a feature of the `CodeSnippet` component itself. Maybe for a follow-up.

Reply:
This is just a simple example of how dropdown in code snippet can be used. But there may be other uses. One dropdown may affect multiple code blocks (input and output), or (with next version of Vanilla) there may be multiple dropdowns on a single code block, they may or may not affect each other. It would be really hard to implement all the possible interactions in CodeSnippet itself. Especially, that we currently don't use these dropdowns in products, so it's hard to say what real uses will be.

So I wanted to focus on creating stateless 'visual' component, that lets you render whatever code snippet you need, but leave the implementation of stateful controller to products.

But I agree, that in future we maybe should have a stateful implementation of code snippet with dropdowns functionality, once some consistent use case appears across products.